modules/globebrowsing/util/globelabelscomponent.h

```diff
@@ -0,0 +1,133 @@
+* Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+* software and associated documentation files (the "Software"), to deal in the Software *
+* without restriction, including without limitation the rights to use, copy, modify,    *
+* merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+* permit persons to whom the Software is furnished to do so, subject to the following   *
+* conditions:                                                                           *
+*                                                                                       *
+* The above copyright notice and this permission notice shall be included in all copies *
+* or substantial portions of the Software.                                              *
+*                                                                                       *
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+* INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+* PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+* HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+* CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+* OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+****************************************************************************************/
+
+#ifndef __OPENSPACE_MODULE_GLOBEBROWSING___GLOBELABELSCOMPONENT___H__
+#define __OPENSPACE_MODULE_GLOBEBROWSING___GLOBELABELSCOMPONENT___H__
+
+#include <openspace/properties/propertyowner.h>
+
+#include <openspace/properties/scalar/boolproperty.h>
+#include <openspace/properties/scalar/floatproperty.h>
+#include <openspace/properties/scalar/intproperty.h>
+#include <openspace/properties/vector/vec4property.h>
+#include <openspace/properties/optionproperty.h>
+
+#include <ghoul/glm.h>
+#include <ghoul/font/fontrenderer.h>
+
+namespace ghoul { class Dictionary;}
+
+namespace ghoul::opengl { class ProgramObject; }
+
+namespace openspace {
+    struct RenderData;
+    
+    namespace documentation { struct Documentation; }
+    namespace globebrowsing { class RenderableGlobe; }
+
+    class GlobeLabelsComponent : public properties::PropertyOwner {
+    public:
+        enum labelRenderingAlignmentType {
+            Horizontally = 0,
+            Circularly = 1
+        };
+
+        // Labels Structures
+        struct LabelEntry {
+            char feature[256];
+            float diameter;
+            float latitude;
+            float longitude;
+            glm::vec3 geoPosition;
+        };
+        struct Labels {
+            std::string filename;
+            std::vector<LabelEntry> labelsArray;
+        };
+
+        const double labelFadeRangeConst = 1500.0;
```

Comment:
Do these have to live in the header file or can the constants be moved to the cpp?
Better to keep the header as simple as possible